data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -19,10 +19,21 @@
 
     private AadAuthenticationHelper aadAuthenticationHelper;
     private String clusterUrl;
+    private String originApplicationVersion;
 
     public ClientImpl(ConnectionStringBuilder csb) throws URISyntaxException {
         clusterUrl = csb.getClusterUrl();
         aadAuthenticationHelper = new AadAuthenticationHelper(csb);
+        if(csb.originApplicationVersion() == null){
+            String version = Utils.class.getPackage().getImplementationVersion();
```

Comment:
What is that? Utils.class.getPackage().getImplementationVersion();

Reply:
its exactly what was used before in Utils.java 
its looking at the version in the module manifest